src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.hpp

```diff
@@ -0,0 +1,258 @@
+    using type = ::Domain<Dim, Frame::Inertial>;
+  };
+
+  using AddOptionsToDataBox =
+      Parallel::ForwardAllOptionsToDataBox<tmpl::list<InitialExtents, Domain>>;
+  using compute_tags =
+      db::AddComputeTags<Tags::MinimumGridSpacing<Dim, Frame::Inertial>>;
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ParallelComponent,
+            Requires<tmpl::list_contains_v<DbTagsList, Domain>> = nullptr>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ElementIndex<Dim>& array_index,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto initial_extents = db::get<InitialExtents>(box);
+    ::Domain<Dim, Frame::Inertial> domain{};
+    db::mutate<Domain>(
+        make_not_null(&box), [&domain](const auto domain_ptr) noexcept {
+          domain = std::move(*domain_ptr);
+        });
+
+    return std::make_tuple(
+        db::create_from<db::RemoveTags<>,
+                        db::AddSimpleTags<>,
+                        compute_tags>(
+            elliptic::Initialization::Domain<Dim>::initialize(
+                db::create_from<typename AddOptionsToDataBox::simple_tags>(
+                    std::move(box)),
+                array_index, initial_extents, domain)),
+        true);
+  }
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ParallelComponent,
+            Requires<not tmpl::list_contains_v<DbTagsList, Domain>> = nullptr>
+  static std::tuple<db::DataBox<DbTagsList>&&, bool> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+      const ElementIndex<Dim>& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    return {std::move(box), true};
+  }
+};
+
+struct PrintMinimumGridSpacing {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const double& overall_min_grid_spacing) noexcept {
+    Parallel::printf("Overall inertial minimum grid spacing: %f\n\n",
+              overall_min_grid_spacing);
+  }
+};
+
+template <size_t Dim>
+struct ArrayReduce {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex,
+            Requires<tmpl::list_contains_v<DbTags, Tags::Mesh<Dim>>> = nullptr>
```

Comment:
When you make this an iterable action, you won't need the `Requires` anymore. That's because we know the DataBox has the mesh in it because it was added in a preceding phase.